octavia-cli/octavia_cli/check_context.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import os
+
+import airbyte_api_client
+import click
+from airbyte_api_client.api import health_api, workspace_api
+from airbyte_api_client.model.workspace_id_request_body import WorkspaceIdRequestBody
+from urllib3.exceptions import MaxRetryError
+
+from .init.commands import DIRECTORIES_TO_CREATE as REQUIRED_PROJECT_DIRECTORIES
+
+
+class UnhealthyApiError(click.ClickException):
+    pass
+
+
+class UnreachableAirbyteInstanceError(click.ClickException):
+    pass
+
+
+class WorkspaceIdError(click.ClickException):
+    pass
+
+
+def check_api_health(api_client: airbyte_api_client.ApiClient) -> None:
+    """Check if the Airbyte API is network reachable and healthy.
+
+    Args:
+        api_client (airbyte_api_client.ApiClient): Airbyte API client.
+
+    Raises:
+        click.ClickException: Raised if the Airbyte api server is unavailable according to the API response.
+        click.ClickException: Raised if the Airbyte URL is not reachable.
+    """
+    api_instance = health_api.HealthApi(api_client)
+    try:
+        api_response = api_instance.get_health_check()
+        if not api_response.available:
+            raise UnhealthyApiError("Your Airbyte instance is not ready to receive requests.")
+    except (airbyte_api_client.ApiException, MaxRetryError):
+        raise UnreachableAirbyteInstanceError(
+            "Could not reach your Airbyte instance, make sure the instance is up and running an network reachable."
```

Comment:
It could be helpful to include the error message of the exception that was thrown here, for debugging purposes